src/place-screen/components/EstimatedCallItem.tsx

```diff
@@ -154,49 +116,66 @@ export function EstimatedCallItem({
               />
             )}
             <ThemeText style={styles.lineName} testID={testID + 'Name'}>
-              {departure.destinationDisplay?.frontText}
+              {departure.destinationDisplay?.frontText ?? ''}
             </ThemeText>
           </View>
-          {mode === 'Departure' || mode === 'Map' ? (
+          {aimedTime && expectedTime && isRealtime !== undefined ? (
             <DepartureTime
-              isRealtime={departure.realtime}
-              isTripCancelled={isTripCancelled}
-              expectedTime={departure.expectedDepartureTime}
-              aimedTime={departure.aimedDepartureTime}
+              isRealtime={isRealtime}
+              isTripCancelled={departure.cancellation}
+              expectedTime={expectedTime}
+              aimedTime={aimedTime}
               testID={testID}
+              now={now}
             />
           ) : null}
-          {allowFavouriteSelection && (
+          {showFavorite && (
             <FavouriteDepartureToggle
               existingFavorite={existingFavorite}
               onMarkFavourite={
-                mode === 'Departure' ? onMarkFavourite : undefined
-              }
-              toggleFavouriteAccessibilityLabel={
-                mode === 'Departure'
-                  ? toggleFavouriteAccessibilityLabel
+                onPressFavorite
+                  ? () => onPressFavorite(departure, existingFavorite)
                   : undefined
               }
+              toggleFavouriteAccessibilityLabel={favoriteAccessibilityLabel}
             />
           )}
         </View>
       </PressableOpacity>
-    </PressableOpacityOrView>
-  );
-}
+    );
+  },
+  (prev, next) => {
+    if (prev.now.valueOf() !== next.now.valueOf()) return false;
+    if (prev.departure.serviceJourney.id !== next.departure.serviceJourney.id)
+      return false;
+    if (prev.departure.aimedDepartureTime !== next.departure.aimedDepartureTime)
+      return false;
+    if (prev.accessibilityHint !== next.accessibilityHint) return false;
+    if (prev.onPress !== next.onPress) return false;
+    if (prev.showFavorite !== next.showFavorite) return false;
+    if (prev.existingFavorite?.id !== next.existingFavorite?.id) return false;
+    if (prev.favoriteAccessibilityLabel !== next.favoriteAccessibilityLabel)
+      return false;
+    if (prev.onPressFavorite !== next.onPressFavorite) return false;
+    if (prev.showNotices !== next.showNotices) return false;
+    return true;
```

Comment:
It would be nice if this can be refactored or moved it to his own checker.

Reply:
Since there are quite a few things that have changed since you suggested this, do you still think it should be refactored? We could always move the `(prev, next) => {}` block to it's own function, but to me this is just as readable 🤷